pkg/detector/policy.go

```diff
@@ -0,0 +1,118 @@
+package detector
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+)
+
+func (d *ResourceDetector) propagateResource(object *unstructured.Unstructured, objectKey keys.ClusterWideKey) error {
+	// 1. attempt to find the matched PropagationPolicy.
+	policyLabels := object.GetLabels()
+	claimedNamespace := util.GetLabelValue(policyLabels, policyv1alpha1.PropagationPolicyNamespaceLabel)
+	claimedName := util.GetLabelValue(policyLabels, policyv1alpha1.PropagationPolicyNameLabel)
+	if claimedNamespace != "" && claimedName != "" {
+		return d.reapplyPropagationPolicy(object, objectKey, claimedNamespace, claimedName)
```

Comment:
`reapply`, do we have other alternatives for the naming?

Reply:
How about `applyPolicyWithNamespaceName` and `applyClusterPolicyWithName`.